homeassistant/components/scrape/config_flow.py

```diff
@@ -95,35 +95,54 @@
     vol.Optional(CONF_ENCODING, default=DEFAULT_ENCODING): TextSelector(),
 }
 
+NONE_SENTINEL = "none"
+
 SENSOR_SETUP = {
     vol.Required(CONF_SELECT): TextSelector(),
     vol.Optional(CONF_INDEX, default=0): NumberSelector(
         NumberSelectorConfig(min=0, step=1, mode=NumberSelectorMode.BOX)
     ),
     vol.Optional(CONF_ATTRIBUTE): TextSelector(),
     vol.Optional(CONF_VALUE_TEMPLATE): TemplateSelector(),
-    vol.Optional(CONF_DEVICE_CLASS): SelectSelector(
+    vol.Required(CONF_DEVICE_CLASS): SelectSelector(
         SelectSelectorConfig(
-            options=[cls.value for cls in SensorDeviceClass],
+            options=[NONE_SENTINEL]
+            + sorted(
+                [
+                    cls.value
+                    for cls in SensorDeviceClass
+                    if cls != SensorDeviceClass.ENUM
+                ]
+            ),
             mode=SelectSelectorMode.DROPDOWN,
+            translation_key="device_class",
         )
     ),
-    vol.Optional(CONF_STATE_CLASS): SelectSelector(
+    vol.Required(CONF_STATE_CLASS): SelectSelector(
         SelectSelectorConfig(
-            options=[cls.value for cls in SensorStateClass],
+            options=[NONE_SENTINEL] + sorted([cls.value for cls in SensorStateClass]),
             mode=SelectSelectorMode.DROPDOWN,
+            translation_key="state_class",
         )
     ),
-    vol.Optional(CONF_UNIT_OF_MEASUREMENT): SelectSelector(
+    vol.Required(CONF_UNIT_OF_MEASUREMENT): SelectSelector(
         SelectSelectorConfig(
-            options=[cls.value for cls in UnitOfTemperature],
+            options=[NONE_SENTINEL] + sorted([cls.value for cls in UnitOfTemperature]),
             custom_value=True,
             mode=SelectSelectorMode.DROPDOWN,
+            translation_key="unit_of_measurement",
         )
     ),
 }
 
 
+def _strip_sentinel(options: dict[str, Any]) -> None:
+    """Convert sentinel to None."""
+    for key in (CONF_DEVICE_CLASS, CONF_STATE_CLASS, CONF_UNIT_OF_MEASUREMENT):
+        if options.get(key) == NONE_SENTINEL:
+            options.pop(key, None)
```

Comment:
We've already checked the key is there and has the sentinel value, either pop it with no default or del it.